File: python/svix/models/snowflake_config.py
# this file is @generated
import typing as t

from .common import BaseModel


class SnowflakeConfig(BaseModel):
    """Configuration parameters for defining a Snowflake sink."""

    private_key: str
    """PEM-encoded private key used for signing token-based requests to the Snowflake API.

    Beginning/end delimiters are not required."""

    account_identifier: str
    """Snowflake account identifier, which includes both the organization and account IDs separated by a hyphen."""

    user_id: str
    """The Snowflake user id."""

    db_name: t.Optional[str] = None
    """Database name.

    Only required if not using transformations."""

    schema_name: t.Optional[str] = None
    """Schema name.

    Only required if not using transformations."""

    table_name: t.Optional[str] = None
    """Table name.

    Only required if not using transformations."""

File: kotlin/lib/src/main/kotlin/models/SnowflakeConfig.kt
// This file is @generated
package com.svix.kotlin.models

import kotlinx.serialization.Serializable

@Serializable
data class SnowflakeConfig(
    /**
     * Snowflake account identifier, which includes both the organization and account IDs separated
     * by a hyphen.
     */
    val accountIdentifier: String,
    /**
     * Database name.
     *
     * Only required if not using transformations.
     */
    val dbName: String? = null,
    /**
     * PEM-encoded private key used for signing token-based requests to the Snowflake API.
     *
     * Beginning/end delimiters are not required.
     */
    val privateKey: String,
    /**
     * Schema name.
     *
     * Only required if not using transformations.
     */
    val schemaName: String? = null,
    /**
     * Table name.
     *
     * Only required if not using transformations.
     */
    val tableName: String? = null,
    /** The Snowflake user id. */
    val userId: String,
)
